applications/asset_tracker_v2/src/cloud/cloud_codec/lwm2m/lwm2m_codec_helpers.c

```diff
@@ -0,0 +1,1043 @@
+	err = lwm2m_engine_set_res_buf(LWM2M_PATH(LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID,
+							0, AVAIL_NETWORK_BEARER_ID, 1),
+					&bearers[1], sizeof(bearers[1]), sizeof(bearers[1]),
+					LWM2M_RES_DATA_FLAG_RO);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_res_buf(LWM2M_PATH(LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID,
+							0, IP_ADDRESSES, 0),
+					modem_dynamic->ip, sizeof(modem_dynamic->ip),
+					sizeof(modem_dynamic->ip), LWM2M_RES_DATA_FLAG_RO);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_res_buf(LWM2M_PATH(LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID,
+							0, APN, 0),
+					modem_dynamic->apn, sizeof(modem_dynamic->apn),
+					sizeof(modem_dynamic->apn), LWM2M_RES_DATA_FLAG_RO);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_s8(LWM2M_PATH(LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID, 0, RSS),
+				  (int8_t)modem_dynamic->rsrp);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_u32(LWM2M_PATH(LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID, 0, CELLID),
+				   modem_dynamic->cell);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_u16(LWM2M_PATH(LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID, 0, SMNC),
+				   modem_dynamic->mnc);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_u16(LWM2M_PATH(LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID, 0, SMCC),
+				   modem_dynamic->mcc);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_u16(LWM2M_PATH(LWM2M_OBJECT_CONNECTIVITY_MONITORING_ID, 0, LAC),
+				   modem_dynamic->area);
+	if (err) {
+		return err;
+	}
+
+	err = date_time_now(&current_time);
+	if (err) {
+		LOG_WRN("Failed getting current time, error: %d", err);
+		return err;
+	}
+
+	err = lwm2m_engine_set_time(LWM2M_PATH(LWM2M_OBJECT_DEVICE_ID, 0, CURRENT_TIME_RID),
+				    (current_time / MSEC_PER_SEC));
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int lwm2m_codec_helpers_set_modem_static_data(struct cloud_data_modem_static *modem_static)
+{
+	int err;
+
+	if (!modem_static->queued) {
+		return -ENODATA;
+	}
+
+	err = lwm2m_engine_set_res_buf(LWM2M_PATH(LWM2M_OBJECT_DEVICE_ID, 0, MODEL_NUMBER_RID),
+				       CONFIG_BOARD,
+				       sizeof(CONFIG_BOARD),
+				       sizeof(CONFIG_BOARD),
+				       LWM2M_RES_DATA_FLAG_RO);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_res_buf(LWM2M_PATH(LWM2M_OBJECT_DEVICE_ID, 0, HARDWARE_VERSION_RID),
+				       CONFIG_SOC,
+				       sizeof(CONFIG_SOC),
+				       sizeof(CONFIG_SOC),
+				       LWM2M_RES_DATA_FLAG_RO);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_res_buf(LWM2M_PATH(LWM2M_OBJECT_DEVICE_ID, 0, MANUFACTURER_RID),
+				       CONFIG_CLOUD_CODEC_MANUFACTURER,
+				       sizeof(CONFIG_CLOUD_CODEC_MANUFACTURER),
+				       sizeof(CONFIG_CLOUD_CODEC_MANUFACTURER),
+				       LWM2M_RES_DATA_FLAG_RO);
+	if (err) {
+		return err;
+	}
+
+	err = lwm2m_engine_set_res_buf(LWM2M_PATH(LWM2M_OBJECT_DEVICE_ID, 0, FIRMWARE_VERSION_RID),
+				       modem_static->appv,
+				       sizeof(modem_static->appv),
```

Comment:
Perhaps `strnlen` should be used here instead to avoid copying in random bytes after null termination

Reply:
Changed all occurrences to use `strlen` instead.